src/python/gudhi/simplex_tree.py

```diff
@@ -0,0 +1,456 @@
+            "Since Gudhi 3.2, calling SimplexTree.initialize_filtration is unnecessary.",
+            DeprecationWarning,
+        )
+        super().initialize_filtration()
+
+    @staticmethod
+    def create_from_array(filtrations, max_filtration: float = float("inf")):
+        """Creates a new, empty complex and inserts vertices and edges. The vertices are numbered from 0 to n-1, and
+        the filtration values are encoded in the array, with the diagonal representing the vertices. It is the
+        caller's responsibility to ensure that this defines a filtration, which can be achieved with either::
+
+            filtrations[np.diag_indices_from(filtrations)] = filtrations.min(axis=1)
+
+        or::
+
+            diag = filtrations.diagonal()
+            filtrations = np.fmax(np.fmax(filtrations, diag[:, None]), diag[None, :])
+
+        :param filtrations: the filtration values of the vertices and edges to insert. The matrix is assumed to be symmetric.
+        :type filtrations: numpy.ndarray of shape (n,n)
+        :param max_filtration: only insert vertices and edges with filtration values no larger than max_filtration
+        :type max_filtration: float
+        :returns: the new complex
+        :rtype: SimplexTree
+        """
+        # TODO: document which half of the matrix is actually read?
+        filtrations = np.asanyarray(filtrations, dtype=float)
+        assert (
+            filtrations.shape[0] == filtrations.shape[1]
+        ), "create_from_array() expects a square array"
+        ret = SimplexTree()
+        ret.insert_matrix(filtrations, max_filtration)
+        return ret
+
+    def insert_edges_from_coo_matrix(self, edges):
+        """Inserts edges given by a sparse matrix in `COOrdinate format
+        <https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.coo_matrix.html>`_.
+        If an edge is repeated, the smallest filtration value is used. Missing entries are not inserted.
+        Diagonal entries are currently interpreted as vertices, although we do not guarantee this behavior
+        in the future, and this is only useful if you want to insert vertices with a smaller filtration value
+        than the smallest edge containing it, since vertices are implicitly inserted together with the edges.
+
+        :param edges: the edges to insert and their filtration values.
+        :type edges: scipy.sparse.coo_matrix of shape (n,n)
+
+        .. seealso:: :func:`insert_batch`
+        """
+        # Without this, it could be slow if we end up inserting vertices in a bad order (flat_map).
+        super().insert_batch_vertices(
+            np.unique(np.stack((edges.row, edges.col))), float("inf")
+        )
+        # TODO: optimize this?
+        for edge in zip(edges.row, edges.col, edges.data):
+            super().insert((edge[0], edge[1]), edge[2])
+
+    def insert_batch(
+        self,
+        vertex_array: Union[np.ndarray[np.float32], np.ndarray[np.float64]],
```

Comment:
Why the recent int -> float?
By the way, I am fine with dropping support for Python 3.8. PyTorch requires 3.9+, SciPy requires 3.10+, https://scientific-python.org/specs/spec-0000/ recommends only supporting 3.11+ for packages released now.

Reply:
Thanks, it was a bad copy/paste. Fixed on [78ed941](https://github.com/GUDHI/gudhi-devel/pull/1197/commits/78ed941829dbdb8b7b1b2e8df405d3d6f6c1c2eb)